custom/dfci_swasth/rules/custom_actions.py

```diff
@@ -0,0 +1,81 @@
+from corehq.apps.data_interfaces.models import CaseRuleActionResult
+from corehq.apps.hqcase.utils import update_case, AUTO_UPDATE_XMLNS
+from corehq.form_processor.exceptions import CaseNotFound
+from corehq.form_processor.models import CommCareCase
+from custom.dfci_swasth.constants import (
+    CASE_TYPE_PATIENT,
+    PROP_CCUSER_CASELOAD_CASE_ID,
+    PROP_COUNSELLOR_CLOSED_CASE_LOAD,
+    PROP_COUNSELLOR_LOAD,
+)
+
+
+def update_counsellor_load(patient_case, rule):
+    num_related_updates = 0
+
+    if patient_case.type != CASE_TYPE_PATIENT:
+        return CaseRuleActionResult(num_related_updates=num_related_updates)
+
+    ccuser_caseload_case = _get_ccuser_caseload_case(patient_case)
+
+    # Updating the case type 'ccuser_caseload' case properties
+    if ccuser_caseload_case:
+        case_updates = _get_case_updates(ccuser_caseload_case)
+        if case_updates:
+            (submission, cases) = update_case(
+                ccuser_caseload_case.domain,
+                ccuser_caseload_case.case_id,
+                case_properties=case_updates,
+                xmlns=AUTO_UPDATE_XMLNS,
+                device_id=__name__ + ".update_counsellor_load",
+                form_name=rule.name,
+            )
+            num_related_updates = 1
+            rule.log_submission(submission.form_id)
+
+    return CaseRuleActionResult(
+        num_related_updates=num_related_updates,
+    )
+
+
+def _get_ccuser_caseload_case(patient_case):
+    coun_ccuser_caseload_case_id = patient_case.get_case_property(PROP_CCUSER_CASELOAD_CASE_ID)
```

Comment:
nit: Could rename `coun_ccuser_caseload_case_id` to simply `case_id` since there is just one `case_id` var in this function

Reply:
Changed
https://github.com/dimagi/commcare-hq/compare/2fb18c32d0b1fbf2d7e7d8c79a5e43753f7f7433..b4454ac55a76b9d95340aae30b0561cae6bdbf6a